src/components/Swap/SwapActionButton/SwapButton.tsx

```diff
@@ -90,7 +90,9 @@ export default function SwapButton({ disabled }: { disabled: boolean }) {
       })
 
       // Only close the review modal if the swap submitted (ie no-throw).
-      setOpen(false)
+      if (submitted) {
```

Comment:
I think @infredible has an opinion (for a future design) on what the UX should be for user rejection, it may be good to check in.